### PR TITLE
[addons] change audioencoder interface to C++ style

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
This is the next addon type where becomes changed to new style.

The changes are nearly equal like on the others.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                       int inRate,
+                       int inBits,
+                       std::string title,
+                       std::string artist,
+                       std::string albumartist,
+                       std::string album,
+                       std::string year,
+                       std::string track,
+                       std::string genre,
+                       std::string comment,
```
Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h]: by-value or by-reference? </code>std::string</code> vs. <code>const std::string&</code>?

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    /// \brief Encode a chunk of audio (**required**)
+    ///
+    /// \param[in] numBytesRead         Number of bytes in input buffer
+    /// \param[in] pbtStream            the input buffer
+    /// \return                         Number of bytes consumed
+    ///
+    virtual int Encode(int numBytesRead, const uint8_t* pbtStream)=0;
```
Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h]: Please add blacks around "=" => <code>) = 0;</code>

File: system/settings/settings.xml [hunk 3]
```diff
@@ -557,7 +557,7 @@
         </setting>
         <setting id="audiocds.encoder" type="addon" label="621" help="36287">
           <level>3</level>
-          <default>audioencoder.xbmc.builtin.wav</default>
+          <default>audioencoder.xbmc.builtin.aac</default>
```
Comment (FernetMenta) [inline system/settings/settings.xml]: btw: builtin encoders addons should be dropped in favour of ffmpeg

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h (hunk 2 above)
Comment (PeterPablo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h]: blanks

Comment (ksooo) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioEncoder.h]: Haha

**Review by ksooo — APPROVED** [review]
Good to go, imo.

**Comment by AlwinEsch** [comment]
On last commit becomes the addon type name changed to "kodi.audioencoder"

@MartijnKaijser is this OK for you?

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**